notebooks/03-query/07-functions-datetime.ipynb

```diff
@@ -0,0 +1,963 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9630abff-f406-4cf6-b846-f03145d40a30",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  __time AS \"start\",\n",
+    "  MILLIS_TO_TIMESTAMP(\"ended_at\") AS \"end\"\n",
+    "FROM \"example-koalas-fndatetime\"\n",
+    "WHERE TIME_IN_INTERVAL(MILLIS_TO_TIMESTAMP(\"ended_at\"),'2019-08-25T13/PT1M')\n",
+    "AND TIME_IN_INTERVAL(__time, 'PT1M/2019-08-25T13')\n",
+    "LIMIT 10\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "510f208b-fdaa-45fe-9908-c6ebbe4c754c",
+   "metadata": {},
+   "source": [
+    "## Use functions to format the date and time\n",
+    "\n",
+    "Below are examples of:\n",
+    "\n",
+    "* Using `TIME_FORMAT` to apply a string pattern to a timestamp.\n",
+    "* Extracting elements of a timestamp using `TIME_EXTRACT` and `EXTRACT`.\n",
+    "\n",
+    "The `TIME_FORMAT` function applies a [JODA](https://www.joda.org/joda-time/apidocs/org/joda/time/format/DateTimeFormat.html) picture to a datetime column.\n",
+    "\n",
+    "Run the cell below to see a number of examples."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ca3165ca-d5c1-4023-8ca1-9cb8bb203bee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  TIME_FORMAT(__time, 'hh:mm:ss') AS \"sessionStart-HMS\",\n",
+    "  TIME_FORMAT(CURRENT_TIMESTAMP, 'YYYY-MM') AS \"now-YM\",\n",
+    "  TIME_FORMAT(TIME_SHIFT(MILLIS_TO_TIMESTAMP(\"ended_at\"),'PT1H',3), 'YYYY-MM-dd (DD)') AS \"ended_at-plus3Hours-YMD-DOY\",\n",
+    "  TIME_FORMAT(TIME_SHIFT(__time,'P1Y',-5), 'EE dd MMM YY:hh a z') AS \"sessionStart-minus5Years-reallyPretty\"\n",
+    "FROM \"example-koalas-fndatetime\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T13/PT1S')\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd09239e-0a6d-407a-85ac-e100cd527924",
+   "metadata": {},
+   "source": [
+    "Using `EXTRACT` and `TIME_EXTRACT`, particular portions of datetimes are returned."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "adb771c1-c52e-42c3-b4c3-b47d9919554f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  EXTRACT(HOUR FROM __time) AS \"start-hour\",\n",
+    "  EXTRACT(HOUR FROM MILLIS_TO_TIMESTAMP(\"ended_at\")) AS \"end-hour\",\n",
+    "  COUNT(*) AS \"sessions\"\n",
+    "FROM \"example-koalas-fndatetime\"\n",
+    "GROUP BY 1, 2\n",
+    "'''\n",
+    "\n",
+    "df = pd.DataFrame(sql_client.sql(sql))\n",
+    "df_group=df.groupby(['start-hour','end-hour']).sum().unstack()\n",
+    "df_group.plot.bar(stacked=\"true\", legend=None)"
```

Comment:
why no legend?

Reply:
I just felt it looked a bit cluttered when it generated...